### PR TITLE
HOCS-5271: Move testing to GitHub actions

### PR DESCRIPTION
**Pull request by nickmcmahon01**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -1,76 +1,37 @@
...
 on:
-  push:
-    branches: [ main ]
   pull_request:
-    # The branches below must be a subset of the branches above
-    branches: [ main ]
+    types: [ opened, reopened, synchronize ]
+    paths:
+      - 'src/main/java/**'
   schedule:
```
Comment (LiamMacP) [inline .github/workflows/codeql-analysis.yml:7]: Do we need this if we are doing it on a pull_request?

File: .drone.yml [hunk 2]
```diff
@@ -1,73 +1,3 @@
...
-  - name: test project
-    image: quay.io/ukhomeofficedigital/hocs-base-image-build:1.0.2
-    environment:
-      LOCALSTACK_HOST: "localstack"
-    commands:
-      - ./gradlew clean build --no-daemon
-    depends_on:
-      - setup localstack
-
-  - name: json lint
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply .drone.yml:41]: we have it in a separate drone tree file, that can be a separate ticket, it still happens.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (nickmcmahon01) [reply .github/workflows/codeql-analysis.yml:7]: the schedule? its for services that aren't under much active development so we still get new alerts

File: .github/workflows/docker-build-tag.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: SemVer tag the commit
+        uses: UKHomeOffice/semver-tag-action@v2
+        with:
+          increment: ${{ steps.label.outputs.matchedLabels }}
```
Comment (LiamMacP) [inline .github/workflows/docker-build-tag.yml:57]: Can we create a ticket for what was discussed about making this allow a single value please?

File: .github/workflows/docker-build-tag.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+      - name: Build container
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: SemVer tag the commit
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply .github/workflows/docker-build-tag.yml:54]: open to suggestions

File: build.gradle [hunk 5]
```diff
@@ -1,13 +1,11 @@
-import org.gradle.api.JavaVersion
```
Comment (LiamMacP) [inline build.gradle:1]: Is thiss not required for `sourceCompatibility`?

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (LiamMacP) [reply .github/workflows/codeql-analysis.yml:7]: Yeah I didn't think there would be any new alerts - but I suppose its better than not having it.

File: .drone.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (LiamMacP) [reply .drone.yml:41]: Yeah I'm happy with that.

File: build.gradle (hunk 5 above)
Comment (nickmcmahon01) [reply build.gradle:1]: green tick 🤷

Comment (LiamMacP) [reply build.gradle:1]: Can you spin it up locally just to make sure please?

File: .github/workflows/docker-build-tag.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (LiamMacP) [reply .github/workflows/docker-build-tag.yml:54]: `Tag repository with SemVer` it just doesn't read right at the minute.

File: build.gradle (hunk 5 above)
Comment (nickmcmahon01) [reply build.gradle:1]: we do, in the integration tests.

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Use Java 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
```
Comment (LiamMacP) [inline .github/workflows/test.yml:18]: I do wonder if adding the `cache: 'gradle'` is worth it here.

**Review by LiamMacP — APPROVED** [review]
Looks fine - one small thing that can be done seperately.

File: .github/workflows/test.yml (hunk 6 above)
Comment (nickmcmahon01) [reply .github/workflows/test.yml:18]: we can do 
https://github.com/marketplace/actions/gradle-cache
https://github.com/marketplace/actions/docker-layer-caching

**Pull request merged**